### ISSUE TITLE
SSH Analyzer won't trigger proper logging of auth_success for successful "none" method authentications

### ISSUE THREAD

**Issue by telamon99** [bro/bro#192]
Our Bro instance has been firing some SSH::Watched_Country_Login notices where the corresponding SSH log entry does not have auth_success set. That is, our standard tab separated SSH logs are showing auth_success as '-' instead of 'T'.

This seems to be happening only for successful "none" method authentications.

Looking at the SSH Analyzer code (SSH.cc), I think the problem is a missing call to BifEvent::generate_ssh_auth_attempted() in SSH_Analyzer::ProcessEncrypted() when the successful "none" method authentication is detected.

Specifically:

```
		// If our user can authenticate via the "none" method, this
		// packet will be a SSH_MSG_USERAUTH_SUCCESS, which has a
		// fixed (decrypted) size of 8 bytes (1 for content
		// pad-aligned to 8-byte boundaries). relative_len would be
		// -16.
		if ( ! userauth_failure_size && (len + 16 == service_accept_size) )
			{
			auth_decision_made = true;
			if ( ssh_auth_successful )
				BifEvent::generate_ssh_auth_successful(interp->bro_analyzer(), interp->bro_analyzer()->Conn(), true);
			return;
			}
```

should probably become:

```
		// If our user can authenticate via the "none" method, this
		// packet will be a SSH_MSG_USERAUTH_SUCCESS, which has a
		// fixed (decrypted) size of 8 bytes (1 for content
		// pad-aligned to 8-byte boundaries). relative_len would be
		// -16.
		if ( ! userauth_failure_size && (len + 16 == service_accept_size) )
			{
			auth_decision_made = true;
			if ( ssh_auth_attempted )
				BifEvent::generate_ssh_auth_attempted(interp->bro_analyzer(), interp->bro_analyzer()->Conn(), true);
			if ( ssh_auth_successful )
				BifEvent::generate_ssh_auth_successful(interp->bro_analyzer(), interp->bro_analyzer()->Conn(), true);
			return;
			}
```

This matches the structure of the code for when normal successful authentications are detected by:

```
// ...or a success packet.
		if ( len - service_accept_size == -16 )
			{
			auth_decision_made = true;
			if ( ssh_auth_attempted )
				BifEvent::generate_ssh_auth_attempted(interp->bro_analyzer(), interp->bro_analyzer()->Conn(), true);
			if ( ssh_auth_successful )
				BifEvent::generate_ssh_auth_successful(interp->bro_analyzer(), interp->bro_analyzer()->Conn(), false);
			return;
			}
}
```

In bro/scripts/base/protocols/ssh/main.bro, the ssh_auth_attempted() is what sets c$ssh$auth_success and ultimately triggers successful authentications to be logged. If ssh_auth_attempted() is never run, then the SSH log entry is created in connection_state_remove() with c$ssh$auth_success unset.

Since SSH::Watched_Country_Login fires from ssh_auth_successful(), the notice is fired even though the SSH log entry ends up showing auth_success as '-'.

I diagnosed the issue by adding the auth_method_none value from ssh_auth_successful() to the SSH::Watched_Country_Login notice msg and saw that anytime we had an auth_method_none = T, the SSH log entry showed auth_success as '-'.

**Comment by grigorescu** [comment]
Thanks for the detailed submission. I agree with all of this.

I'll work on making a pull request with the proposed changes.

**Comment by grigorescu** [comment]
Fix is in topic/vlad/ssh_auth_none_fix and is ready to be merged.

**Comment by jsiwek** [comment]
Thanks, the bug fix is in master now and will be in final 2.6 release.
